autoPyTorch/pipeline/components/base_component.py

```diff
@@ -224,18 +224,22 @@ def check_requirements(self, X: Dict[str, Any], y: Any = None) -> None:
 
         if y is not None:
             warnings.warn("Provided y argument, yet only X is required")
+        assert 'dataset_properties' in X.keys(), "Fit dictionary must contain dataset_properties"
 
         for requirement in self._fit_requirements:
-            if requirement.name not in X.keys():
+            check_dict = X
```

Comment:
`check_dict = X['dataset_properties'] if requirement.dataset_property else X` might be more readable